Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java

```diff
@@ -0,0 +1,94 @@
+package io.branch.indexing;
+
+import android.content.Context;
+import android.net.Uri;
+import android.text.TextUtils;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+import io.branch.referral.util.LinkProperties;
+
+/**
+ * Created by sojanpr on 5/18/16.
+ * <p>
+ * Helper class for publishing BUO with google app indexing
+ * </p>
+ */
+class AppIndexingHelper {
+    public static void addToAppIndex(final Context context, final BranchUniversalObject buo) {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                String urlForAppIndexing = buo.getShortUrl(context, new LinkProperties());
```

Comment:
let's prefill the link properties similar to iOS: `[linkData setupChannel:@"google_search"];`